src/object.d

```diff
@@ -1908,10 +1908,10 @@ class TypeInfo_Struct : TypeInfo
 
     @safe pure nothrow
     {
-        size_t   function(in void*)           xtoHash;
-        bool     function(in void*, in void*) xopEquals;
-        int      function(in void*, in void*) xopCmp;
-        string   function(in void*)           xtoString;
+        size_t   function(const scope void*)           xtoHash;
+        bool     function(const scope void*, const scope void*) xopEquals;
+        int      function(const scope void*, const scope void*) xopCmp;
+        string   function(const scope void*)           xtoString;
```

Comment:
Pedantically speaking, I don't think those function parameters can be neither `scope` nor `const` :(

Only when the TypeInfo generation is fully template-driven we can have true guarantees about `const`/`scope`/insert your favorite function attribute here correctness.

Reply:
What do you recommend?

Reply:
There are two options:
1. Use honest function signature and change those parameters to `void*` as in reality even the `const`-ness can't be guaranteed. This means that `TypeInfo_Struct.{equals,compare}` must also take `void*` parameters which is likely a breaking change.

2. Continue with the deceit and deal with problems somehow later down the line, when we can be technically honest.

@jmdavis @schveiguy @atilaneves What's your point of view?

Reply:
Not sure. I agree we can't make them `scope` or `const` though.

Reply:
@ZombineDev Why can't these be `const`?  They shouldn't be modifying `p`.

Reply:
@JinShil  Because they are function pointers to arbitrary user code. The pointer itself may not be modified (as it will be copied), but the object pointed by `p` may be. And it can also escape `p` without any issues.

Here's the sort of techniques that we need to use in order to be type safe:
https://gist.github.com/ZombineDev/ac19e93dff4f7b251be0c7af8e8d150d#file-proper_type_info-d-L62-L73

Live demo: https://run.dlang.io/gist/ZombineDev/ac19e93dff4f7b251be0c7af8e8d150d?compiler=dmd&args=-dip1000